fix(bench): preserve workload failure instead of reporting unknown scenario

## Problem

When a `homeboy bench --scenario <id>` workload is discovered successfully but then fails during warmup, Homeboy reports the workload failure and then returns a misleading top-level validation error saying the scenario id is unknown and `discovered ids: <none>`.

Observed while running the Studio BFB bench rig:

```text
homeboy bench --rig studio-bfb --scenario studio-agent-site-build --iterations 1 --ignore-default-baseline

Running Node.js benchmarks...
  Component: studio (/Users/chubes/Developer/studio@bfb-mu-plugin-agent-output)
  Iterations: 1
Discovered 3 Node.js bench scenarios.

Node.js bench run complete.
Running Node.js benchmarks...
  Component: studio (/Users/chubes/Developer/studio@bfb-mu-plugin-agent-output)
  Iterations: 1
WORKLOAD_BEGIN: studio-agent-site-build (studio-agent-site-build.bench.mjs)
WORKLOAD_ERROR: studio-agent-site-build — warmup iteration threw: Studio site-build eval failed: Claude Code returned an error result: Reached maximum number of turns (40); raw_result=/var/folders/.../result-bfb-65596-1777433942904-cfk95ly9t6u.json

============================================
BUILD FAILED: bench-runner.mjs exited with code 1
============================================
{
  "success": false,
  "error": {
    "code": "validation.invalid_argument",
    "message": "Invalid argument 'scenario': unknown bench scenario id(s): studio-agent-site-build; discovered ids: <none>",
    "details": {
      "field": "scenario",
      "id": "studio-agent-site-build",
      "problem": "unknown bench scenario id(s): studio-agent-site-build; discovered ids: <none>",
      "tried": []
    }
  }
}
```

The scenario was not unknown. It was discovered in the first pass. The real failure was the warmup iteration exception from the selected workload.

## Expected behavior

The top-level error should preserve the workload failure as the primary error, e.g.:

```text
bench.workload_failed: scenario studio-agent-site-build failed during warmup iteration 1/1: ...
```

or otherwise surface the bench runner exit as a workload execution failure.

It should not recast the failure as `validation.invalid_argument` / unknown scenario.

## Why it matters

Bench failures become harder to diagnose because the final structured error points at scenario selection rather than the workload exception. This is especially confusing for long-running scenario-specific benches where discovery already printed the selected scenario.

## Acceptance criteria

- A failing selected bench workload produces a workload/runner failure error, not an unknown-scenario validation error.
- The error preserves the failing scenario id and the runner's stderr/stdout failure message.
- Add a regression test for: discovered selected scenario + warmup/workload failure.

## AI assistance

- **AI assistance:** Yes
- **Tool(s):** OpenCode (GPT-5.5)
- **Used for:** Capturing the repro from a live Studio/BFB bench run and drafting the issue.

--- a/src/commands/bench/tests.rs
+++ b/src/commands/bench/tests.rs
@@ -99,6 +99,64 @@ JSON
     }
 }
 
+fn write_failing_bench_extension(home: &TempDir) {
+    let extension_dir = home
+        .path()
+        .join(".config")
+        .join("homeboy")
+        .join("extensions")
+        .join("nodejs");
+    fs::create_dir_all(&extension_dir).expect("mkdir extension");
+    fs::write(
+        extension_dir.join("nodejs.json"),
+        r#"{
+                "name": "Node.js",
+                "version": "0.0.0",
+                "bench": { "extension_script": "bench-runner.sh" }
+            }"#,
+    )
+    .expect("write extension manifest");
+
+    let script_path = extension_dir.join("bench-runner.sh");
+    fs::write(
+        &script_path,
+        r#"#!/bin/sh
+if [ "$HOMEBOY_BENCH_LIST_ONLY" = "1" ]; then
+  cat > "$HOMEBOY_BENCH_RESULTS_FILE" <<JSON
+{
+  "component_id": "$HOMEBOY_COMPONENT_ID",
+  "iterations": 0,
+  "scenarios": [
+    { "id": "studio-agent-site-build", "iterations": 0, "metrics": {} }
+  ]
+}
+JSON
+  exit 0
+fi
+
+cat > "$HOMEBOY_BENCH_RESULTS_FILE" <<JSON
+{
+  "component_id": "$HOMEBOY_COMPONENT_ID",
+  "iterations": ${HOMEBOY_BENCH_ITERATIONS:-0},
+  "scenarios": []
+}
+JSON
+printf 'WORKLOAD_ERROR: studio-agent-site-build - warmup iteration threw: Studio site-build eval failed\n' >&2
+exit 7
+"#,
+    )
+    .expect("write bench script");
+
+    #[cfg(unix)]
+    {
+        let mut permissions = fs::metadata(&script_path)
+            .expect("script metadata")
+            .permissions();
+        permissions.set_mode(0o755);
+        fs::set_permissions(&script_path, permissions).expect("chmod script");
+    }
+}
+
 fn write_registered_component(home: &TempDir, component_id: &str, path: &std::path::Path) {
     let component_dir = home
         .path()
@@ -383,6 +441,42 @@ fn run_selects_single_scenario() {
                 let scenarios = result.results.expect("results").scenarios;
                 assert_eq!(scenarios.len(), 1);
                 assert_eq!(scenarios[0].id, "slow");
+            }
+            _ => panic!("expected single output"),
+        }
+    });
+}
+
+#[test]
+fn selected_scenario_workload_failure_preserves_runner_error() {
+    with_isolated_home(|home| {
+        write_failing_bench_extension(home);
+        let component_dir = tempfile::TempDir::new().expect("component dir");
+        write_registered_component(home, "studio", component_dir.path());
+
+        let (output, exit_code) = run(
+            run_args(
+                Some("studio"),
+                Vec::new(),
+                vec!["studio-agent-site-build".to_string()],
+            ),
+            &GlobalArgs {},
+        )
+        .expect("runner failure should return structured bench output");
+
+        assert_eq!(exit_code, 7);
+        match output {
+            BenchOutput::Single(result) => {
+                assert_eq!(result.status, "failed");
+                assert_eq!(result.exit_code, 7);
+                assert_eq!(result.results.expect("partial results").scenarios.len(), 0);
+                let failure = result.failure.expect("runner failure metadata");
+                assert_eq!(
+                    failure.scenario_id.as_deref(),
+                    Some("studio-agent-site-build")
+                );
+                assert!(failure.stderr_tail.contains("WORKLOAD_ERROR"));
+                assert!(failure.stderr_tail.contains("warmup iteration threw"));
             }
             _ => panic!("expected single output"),
         }

--- a/src/core/extension/bench/run.rs
+++ b/src/core/extension/bench/run.rs
@@ -275,6 +275,33 @@ fn filter_extra_workloads_by_scenario_ids(
         .collect()
 }
 
+fn parse_execution_results_file(
+    results_file: &Path,
+    scenario_ids: &[String],
+    runner_success: bool,
+) -> Result<Option<BenchResults>> {
+    if !results_file.exists() {
+        return Ok(None);
+    }
+
+    if runner_success {
+        return Ok(Some(apply_scenario_filter(
+            parsing::parse_bench_results_file(results_file)?,
+            scenario_ids,
+        )?));
+    }
+
+    Ok(parsing::parse_bench_results_file(results_file).ok())
+}
+
+fn failure_scenario_id(scenario_ids: &[String]) -> Option<String> {
+    if scenario_ids.len() == 1 {
+        Some(scenario_ids[0].clone())
+    } else {
+        None
+    }
+}
+
 fn discover_bench_scenarios(
     execution_context: &ExtensionExecutionContext,
     component: &Component,
@@ -418,14 +445,11 @@ pub fn run_main_bench_workflow(
                 None,
             )?
             .run()?;
-            let parsed = if results_file.exists() {
-                parsing::parse_bench_results_file(&results_file)
-                    .ok()
-                    .map(|results| apply_scenario_filter(results, &execution_args.scenario_ids))
-                    .transpose()?
-            } else {
-                None
-            };
+            let parsed = parse_execution_results_file(
+                &results_file,
+                &execution_args.scenario_ids,
+                runner_output.success,
+            )?;
             let failure_stderr_tail = if !runner_output.success {
                 Some(stderr_tail(&runner_output.stderr))
             } else {
@@ -530,14 +554,14 @@ pub fn run_main_bench_workflow(
     } else {
         baseline_exit_override.unwrap_or(0)
     };
-    let failure = if parsed.is_none() && !runner_success {
+    let failure = if !runner_success {
         failure_stderr_tail.map(|stderr_tail| BenchRunFailure {
             component_id: args.component_id.clone(),
             component_path: args
                 .path_override
                 .clone()
                 .or_else(|| Some(component.local_path.clone())),
-            scenario_id: None,
+            scenario_id: failure_scenario_id(&execution_args.scenario_ids),
             exit_code: runner_exit_code,
             stderr_tail,
         })
@@ -793,14 +817,8 @@ fn run_single_dispatcher(
     }
 
     let runner_output = build_runner(execution_context, component, args, run_dir, None)?.run()?;
-    let parsed = if results_file.exists() {
-        Some(apply_scenario_filter(
-            parsing::parse_bench_results_file(&results_file)?,
-            &args.scenario_ids,
-        )?)
-    } else {
-        None
-    };
+    let parsed =
+        parse_execution_results_file(&results_file, &args.scenario_ids, runner_output.success)?;
     let failure_stderr_tail = if !runner_output.success {
         Some(stderr_tail(&runner_output.stderr))
     } else {

